controllers/grid/users/reviewer/form/ThankReviewerForm.php

```diff
@@ -149,13 +149,16 @@ public function execute(...$functionArgs)
         // update the ReviewAssignment with the acknowledged date
         $newData = ['dateAcknowledged' => Core::getCurrentDate()];
         if (!in_array($reviewAssignment->getConsidered(), [ReviewAssignment::REVIEW_ASSIGNMENT_CONSIDERED, ReviewAssignment::REVIEW_ASSIGNMENT_RECONSIDERED])) {
-            $newData['considered'] = $reviewAssignment->getConsidered() === ReviewAssignment::REVIEW_ASSIGNMENT_NEW
+            $newData['considered'] = ($reviewAssignment->getConsidered() === ReviewAssignment::REVIEW_ASSIGNMENT_NEW ||
+                                     $reviewAssignment->getConsidered() === ReviewAssignment::REVIEW_ASSIGNMENT_VIEWED)
                 ? ReviewAssignment::REVIEW_ASSIGNMENT_CONSIDERED
                 : ReviewAssignment::REVIEW_ASSIGNMENT_RECONSIDERED;
         }
 
-        // set the date when the editor confirms the review
-        $newData['dateConsidered'] = Core::getCurrentDate();
+        if(!$reviewAssignment->getDateConsidered()) {
+            // set the date when the editor confirms the review
+            $newData['dateConsidered'] = Core::getCurrentDate();
+        }
```

Comment:
I noticed while testing, that after thanking it overrode the dateConsidered to currentDate. So also moved it behind condition.